### ISSUE TITLE
Document How to compile HIE with profiling enabled?

### ISSUE THREAD

**Issue by skress** [haskell/haskell-ide-engine#1214]
According to https://github.com/haskell/haskell-ide-engine/issues/556#issuecomment-430202467 I should be able to compile HIE with profiling enabled like this:

```shell
stack install --work-dir=.stack-work-profile --profile
```

I've tried it on a fresh clone of the repository on macOS and Linux and it fails on both systems with the same error:

```
    [34 of 34] Compiling Paths_HaRe       ( .stack-work-profile/dist/x86_64-osx/Cabal-2.4.0.1/build/autogen/Paths_HaRe.hs, .stack-work-profile/dist/x86_64-osx/Cabal-2.4.0.1/build/Paths_HaRe.p_o )
    Preprocessing executable 'ghc-hare' for HaRe-0.8.4.1..
    Building executable 'ghc-hare' for HaRe-0.8.4.1..

    <no location info>: warning: [-Wmissing-home-modules]
        These modules are needed for compilation but not listed in your .cabal file's other-modules:
            Paths_HaRe
    [1 of 2] Compiling Paths_HaRe       ( .stack-work-profile/dist/x86_64-osx/Cabal-2.4.0.1/build/ghc-hare/autogen/Paths_HaRe.hs, .stack-work-profile/dist/x86_64-osx/Cabal-2.4.0.1/build/ghc-hare/ghc-hare-tmp/Paths_HaRe.p_o )
    [2 of 2] Compiling Main             ( app/MainHaRe.hs, .stack-work-profile/dist/x86_64-osx/Cabal-2.4.0.1/build/ghc-hare/ghc-hare-tmp/Main.p_o )
    /Users/skress/work/src/haskell-ide-engine/submodules/HaRe/app/MainHaRe.hs:25:16: fatal:
        cannot find object file ‘.stack-work-profile/dist/x86_64-osx/Cabal-2.4.0.1/build/ghc-hare/ghc-hare-tmp/Paths_HaRe.dyn_o’
        while linking an interpreted expression
```

Any ideas how to avoid this problem? (In case the question arises: I'd like to further debug my problem in #1207 and my current approach to creating a build with profiling seems to be broken.)

**Comment by mpickering** [comment]
Adding `profiling: True` to the `cabal.project` file and using `cabal new-build` worked fine for me when I tried it. I wonder if you might run into problems if you don't have a profiled version of GHC (you have to build this from source so it probably isn't worth the effort). 

Any easier way to debug this issue might be to disable profiling in your project if it's enabled :) ?

**Comment by skress** [comment]
Thanks, kind of worked. For future reference this is what worked for me:

```bash
cat << EOF > cabal.project.local
package *
  profiling: True
EOF

cabal new-install exe:hie --enable-profiling -w /Users/skress/.stack/programs/x86_64-osx/ghc-8.6.4/bin/ghc --write-ghc-environment-files=never --symlink-bindir=$HOME/.local/bin --overwrite-policy=always -j9 --reinstall
```

**Comment by skress** [comment]
@alanz I can give it a try if the approach to build hie with profiling as I described it above is what would like to see in the documentation.

I would add a section to [docs/Build.md#steps-to-build-hie](https://github.com/haskell/haskell-ide-engine/blob/master/docs/Build.md#steps-to-build-hie).

I have tried my approach on macOS and Linux. It worked for a minimal project that I created via "stack new testproject", i.e. in VSCode I could see stacktraces when executing hie with +RTS -xc.

Using that profiling build of hie with a new project created via "stack new yesodtest yesod-sqlite" did not work. I had to rebuild that project itself with profiling support to see stacktraces. Is that expected behavior for projects like Yesod (probably due to it's use of Template Haskell)?
